agent/agent.go

```diff
@@ -64,10 +65,8 @@ func New(opts Options, comm client.Communicator) *Agent {
 // at interval agentSleepInterval and runs them.
 func (a *Agent) Start(ctx context.Context) error {
 	a.startStatusServer(a.opts.StatusPort)
-	err := errors.Wrap(a.loop(ctx), "error in agent loop, exiting")
 	tryCleanupDirectory(a.opts.WorkingDirectory)
-	grip.Emergency(err)
-	return err
+	return errors.Wrap(a.loop(ctx), "error in agent loop, exiting")
```

Comment:
This is now cleaning up the working directory before the agent starts, which sounds like that's how it should have been originally. Was this just not working before?

Reply:
indeed, I think this is a seperate bug than the others, but explains why we weren't cleaning up when we expected.